### DISCUSSION TITLE
PaoPaoDNS（虚拟机docker）+PaoPaoGateWay（虚拟机ISO），白名单走PaoPaoGateWay，force_forward_list.txt中的无法访问，非force_forward_list.txt中的正常

### DISCUSSION THREAD

**Discussion by MyIsZero** [kkkgo/PaoPaoGateWay#81] [问和答]
PaoPaoDNS=192.168.0.250
PaoPaoGateWay=192.168.0.251
```
docker run -d \
--name paopaodns \
-v /root/paopaodns:/data \
--net=host \
-e CNAUTO=yes \
-e SERVER_IP=192.168.0.251 \
-e CUSTOM_FORWARD=192.168.0.251:53 \
-e AUTO_FORWARD=no \
-e ADDINFO=yes \
-e SHUFFLE=yes \
-e TZ=Asia/Shanghai \
-e HTTP_FILE=yes \
-e USE_HOSTS=yes \
--add-host=goog.xyz:108.162.195.140 \
sliamb/paopaodns
```
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/dbf05ce2-578c-4bea-9ba6-7172cc4466cf)
```
[root@archlinux ~]# cat /root/paopaodns/force_forward_list.txt
#focre forward your domains to $CUSTOM_FORWARD DNS server.
#Here are some examples.

# Global bing
# domain:google.com
domain:baidu.com
# domain:bing.com
# domain:bing.net
# domain:msn.com
# domain:msn.cn

# Google play download issues
# domain:googleapis.cn
# domain:xn--ngstr-lra8j.com
# domain:gvt1.com
# domain:android.googleapis.com
# domain:play.googleapis.com
```
```
[root@archlinux ~]# cat /root/paopaodns/ppgw.ini
#paopao-gateway

# mode=socks5|ovpn|yaml|suburl|free
# default: free
mode=free
# 测试模式，能用再切换成socks5或yaml
# Set fakeip's CIDR here
# default: fake_cidr=7.0.0.0/8
fake_cidr=7.0.0.0/8

# Set your trusted DNS here
# default: dns_ip=1.0.0.1
dns_ip=192.168.0.250
# default: dns_port=53
# If used with PaoPaoDNS, you can set the 5304 port
dns_port=5304

# Clash's web dashboard
clash_web_port="80"
clash_web_password="PaoPaoDNS0"

# default：openport=no
# socks+http mixed 1080
openport=no

# default: udp_enable=no
udp_enable=no

# default:30
sleeptime=30

# socks5 mode settting
# default: socks5_ip=gatewayIP
socks5_ip="192.168.0.105"
# default: socks5_port="7890"
socks5_port="10808"

# ovpn mode settting
# The ovpn file in the same directory as the ppgw.ini.
# default: ovpnfile=custom.ovpn
ovpnfile="custom.ovpn"
ovpn_username=""
ovpn_password=""

# yaml mode settting
# The yaml file in the same directory as the ppgw.ini.
# default: yamlfile=custom.yaml
yamlfile="custom.yaml"

# suburl mode settting
suburl="https://..."
subtime=1d

# fast_node=check/yes/no
fast_node=no
test_node_url="https://www.youtube.com/generate_204"
ext_node="Traffic|Expire| GB|Days|Date"
cpudelay="3000"

# dns burn setting
# depend on fast_node=yes & mode=suburl/yaml
dns_burn=no
# If used with PaoPaoDNS, you can set the PaoPaoDNS:53
ex_dns="223.5.5.5:53"

# Network traffic records
net_rec=no
max_rec=5000
```

```
nslookup -type=TXT whoami.ds.akahelp.net 192.168.0.250
服务器:  PaoPaoDNS,blog.03k.org
Address:  192.168.0.250

非权威应答:
whoami.ds.akahelp.net   text =

        "ns"
        "2606:a8c0:3:202::a"

20240330001047.174706656.addinfo.paopaodns      text =

        "Since 0ms From:not_a_aaaa -> forward_unbound"
```
```
nslookup whoami.03k.org 192.168.0.250
服务器:  PaoPaoDNS,blog.03k.org
Address:  192.168.0.250

*** PaoPaoDNS,blog.03k.org 找不到 whoami.03k.org: Server failed
```
```
nslookup -type=TXT www.baidu.com 192.168.0.250
服务器:  PaoPaoDNS,blog.03k.org
Address:  192.168.0.250

www.baidu.com   canonical name = www.a.shifen.com
www.a.shifen.com        canonical name = www.wshifen.com
wshifen.com
        primary name server = ns1.wshifen.com
        responsible mail addr = baidu_dns_master.baidu.com
        serial  = 2403290006
        refresh = 60 (1 min)
        retry   = 30 (30 secs)
        expire  = 2592000 (30 days)
        default TTL = 3600 (1 hour)
20240330001351.954651775.addinfo.paopaodns      text =

        "Since 1605ms From:forward -> 192.168.0.251@53"
```
```
nslookup www.baidu.com 192.168.0.250
服务器:  PaoPaoDNS,blog.03k.org
Address:  192.168.0.250

名称:    www.baidu.com
Address:  7.0.0.18
```
```
nslookup -type=TXT www.bilibili.com 192.168.0.250
服务器:  PaoPaoDNS,blog.03k.org
Address:  192.168.0.250

20240330001500.240.reject.paopaodns     text =

        "536ms, END by f_force_nocn_list."
```
```
nslookup www.bilibili.com 192.168.0.250
服务器:  PaoPaoDNS,blog.03k.org
Address:  192.168.0.250

非权威应答:
名称:    a.w.bilicdn1.com
Addresses:  59.36.228.18
          14.17.92.74
          59.36.228.17
          14.17.92.72
          14.17.92.73
          59.36.228.21
          59.36.228.19
          14.17.92.70
          59.36.228.20
          14.17.92.71
Aliases:  www.bilibili.com
```
```
ping 7.0.0.18

正在 Ping 7.0.0.18 具有 32 字节的数据:
来自 7.0.0.18 的回复: 字节=32 时间=9ms TTL=64
来自 7.0.0.18 的回复: 字节=32 时间=7ms TTL=64
来自 7.0.0.18 的回复: 字节=32 时间=3ms TTL=64
来自 7.0.0.18 的回复: 字节=32 时间=2ms TTL=64

7.0.0.18 的 Ping 统计信息:
    数据包: 已发送 = 4，已接收 = 4，丢失 = 0 (0% 丢失)，
往返行程的估计时间(以毫秒为单位):
    最短 = 2ms，最长 = 9ms，平均 = 5ms
```

**Comment by MyIsZero** [comment]
```
tracert www.baidu.com

通过最多 30 个跃点跟踪
到 www.baidu.com [7.0.0.3] 的路由:

  1    13 ms     1 ms     1 ms  192.168.0.1
  2     3 ms     1 ms     3 ms  7.0.0.3

跟踪完成。
```

**Comment by kkkgo** [comment]
看了上面的debug信息，可能对SERVER_IP参数有误解，这个参数其实是用来映射paopao.dns这个域名到你的提供http 7889端口配置下载服务的主机IP的（也就是DNS所在的docker），不过你网卡信息的DNS是paopaodns提供的话也能获取到配置。然后test的是递归部分失败，看debug是运行没啥问题，可以考虑改善一下nat来改善递归，当然递归不了他也可以回退转发。你开了ADDINFO想看结果来源的话可以用dig命令，在ADDITIONAL SECTION可以看到调试信息，用nslookup查txt记录是无意义的，只是akahelp恰好提供了一个txt记录的服务而已。
以上都是题外话，如果你free模式访问不了，那就要考虑你网关本身的网络问题，因为你目标的域名已经解析成FAKEIP了，说明DNS部分已经完成任务了，你路由也可达了，说明静态路由到核心这一段也完成任务了。
你的镜像是docker定制的还是直接下载的？网卡的IP是dhcp还是静态的？如果是静态的我觉得你可能填错了网关地址导致网络不行。
你可以在ppgw虚拟机内ping 223.5.5.5试试有没有网络，或者traceroute一下也可以。

**Comment by MyIsZero** [comment]
```
docker run -d \
--name paopaodns \
-v /root/paopaodns:/data \
--net=host \
-e CNAUTO=yes \
-e SERVER_IP=192.168.0.250 \
-e CUSTOM_FORWARD=192.168.0.251:53 \
-e AUTO_FORWARD=no \
-e ADDINFO=yes \
-e SHUFFLE=yes \
-e TZ=Asia/Shanghai \
-e HTTP_FILE=yes \
-e USE_HOSTS=yes \
--add-host=goog.xyz:108.162.195.144 \
sliamb/paopaodns
```
```
dig www.baidu.com

; <<>> DiG 9.16.19 <<>> www.baidu.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 5851
;; flags: qr aa rd ra ad; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 2

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; MBZ: 0x0001, udp: 4096
; COOKIE: 33320367f45855e5 (echoed)
;; QUESTION SECTION:
;www.baidu.com.			IN	A

;; ANSWER SECTION:
www.baidu.com.		1	IN	A	7.0.0.3

;; ADDITIONAL SECTION:
20240330101200.936850832.addinfo.paopaodns. 2 IN TXT "Since 1ms From:forward -> 192.168.0.251@53"

;; Query time: 3 msec
;; SERVER: 192.168.0.250#53(192.168.0.250)
;; WHEN: Sat Mar 30 10:12:00 CST 2024
;; MSG SIZE  rcvd: 180
```
docker是中科大拉下来的，没懂什么是改善一下nat，双虚拟机已经都搞静态了，ISO是按文档路由器分配静态的，
其他配置都是抄了一部分，改了一部分来的，今天才发现ISO里面是可以输入命令的，这里面有日志或者配置需要检查吗？
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/8f5c7154-7c8a-44b9-9fe1-5532052fe7ea)
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/a602b789-428c-4aeb-8541-88b5c1f84dda)
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/fc43c589-6172-4d44-a90d-82c100a573b4)
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/03bdd750-f23f-4f1e-a6ee-f3d24a6ae79a)

**Comment by kkkgo** [comment]
你网络是通的话，可以直接到web面板设置里面的日志选debug然后点日志来看，流量到网关的时候就会有日志。没有日志就是流量没有到网关。

**Comment by MyIsZero** [comment]
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/e766976d-2fb0-4789-92d5-9582e8c83c17)
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/683fed3e-b607-4071-805b-c6c948d05fe5)

**Reply by kkkgo** [reply]
没有日志那说明你流量没有到网关，你可以把电脑网卡的网关设置成ppgw的IP，这样相当于电脑直接就到网关，看看有没有日志。

**Comment by MyIsZero** [comment]
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/97e6e157-2ed3-4e5b-a492-e6a3b34c550a)
192.168.0.240/24
192.168.0.251
114.114.114.114
223.5.5.5
网关改ppgw的IP后，无法上网

**Reply by kkkgo** [reply]
你这就是有日志啊，日志也正常，你上面没有日志那是因为你就没有访问你的设置baidu那肯定没有流量过来。
日志调成debug的话你在虚拟机终端也可以看到滚动日志的。
无法上网应该还是虚拟机本身的网络问题，你可以在虚拟机里面`traceroute 223.5.5.5`跟踪下路由。

**Comment by MyIsZero** [comment]
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/edbe280d-8764-4201-b118-31bef2807d2b)
10之后就没返回了，我一直等到30还是***

**Reply by kkkgo** [reply]
足够了，从你这个路由器来看你有两层路由，到了192.168.0.1之后还有192.168.1.1才出去互联网，前面说的改善nat就是减少路由的nat，简单直白来说就是减少路由的层数。nat层数越少递归效果越好。
既然你网络是通的，你可以排除一下是不是设备系统问题。建议用手机这种简单的设备测试，把手机设置成静态IP，DNS和网关都填ppgw来测试，然后观察虚拟机的连接日志。如果有IPV6就先把IPV6关掉。

**Comment by MyIsZero** [comment]
发现一个有点奇怪的地方，
安卓手机网关改PaoPaoGateWay后无法上网，
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/23da088e-9101-43d6-abbf-688806c9c164)
但如果我使用代理，
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/d0c15b87-07be-4313-a214-cb04b96b1dc7)
是可以正常上网的，日志也有正常的显示出来一大堆

**Comment by MyIsZero** [comment]
我吐了，
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/6d368910-3c70-4303-bee8-54f853e55ee0)
开启这个后正常了

**Reply by kkkgo** [reply]
我已经说过了，“把手机设置成静态IP，***DNS*** 和网关都填ppgw”，你是这么设置的话是不需要udp通过也可以打开网页的。如果你设置了其他DNS，你把udp关了DNS流量自然无法出口网关。

**Comment by MyIsZero** [comment]
测单指向网关是正常的，但一旦加入PaoPaoDNS就不行了

**Reply by kkkgo** [reply]
并不是DNS问题，DNS在能按你的预期解析出正确的FAKEIP后就应该是正常工作完成任务。你可以测试手机设置静态IP+DNS:paopaodns+ppgw网关，应该也是预期工作。
你需要检查你的防火墙/路由器/虚拟平台/交换机/静态路由是否预期工作。一些路由器的原生系统可能存在交换机芯片静态路由的bug。
简单的测试方法可以把1.0.0.0/24添加到静态路由到ppgw，然后直接用IP访问比如 https://1.0.0.2/cdn-cgi/trace 。能访问就说明静态路由过去到网关整个过程都是正常的。

**Comment by MyIsZero** [comment]
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/4f36d609-5d40-4b7d-aec7-9d1be8f3432d)
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/77c42bcb-6573-401a-89ae-094c3ccf5dc3)
手机网关直接设置成PaoPaoGateWay是正常访问的，加路由后就上不来了了

**Reply by kkkgo** [reply]
你用free出站当然不正常，这是死循环，你需要一个节点来出站。用这个连接测试是因为可以看到你当前的出口IP（同时排除DNS问题，因为这个连接地址不需要DNS）。

**Reply by kkkgo** [reply]
free模式意义在于给他分配了不同的网关，或者在路由器上做了策略路由他的出口线路/物理线路不一样。又或者可以当成局域网抓包工具使用（把对应设备手动设置网关）。
相同的网关的情况下，设置静态路由下一跳到ppgw，ppgw回到路由器下一跳仍然是ppgw，这就是死循环。当然，如果是经过FAKEIP池映射的话，这个过程不会有什么问题，到达ppgw之后会还原真实IP出站。如果把真实IP添加到静态路由可能就会有问题了。

**Reply by MyIsZero** [reply]
```
#paopao-gateway

# mode=socks5|ovpn|yaml|suburl|free
# default: free
mode=socks5
```
设置成 这个，
```
https://1.0.0.2/cdn-cgi/trace/
```
，正常访问了，在路由后台关闭静态和启动静态路由，可以观察到这个网站上的
ip=27. 变到，ip=66. 
当我启动静态路由时，ip=66.，跟我手机把网关直接指向PaoPaoGateWay的结果是一样的，当我关闭静态路由时会变回ip=27. 
但访问force_forward_list.txt中的域名还是一样，通不了

**Reply by kkkgo** [reply]
当你的目标域名成功解析成FAKEIP，并静态路由到PPGW，但却无法访问，那么你就需要考虑，有没有可能是你的节点禁止了访问某些网站。
比如force_forward添加：`domain:cloudflare.com`
终端设备手动dig/nslookup www.cloudflare.com, 看是否解析为FAKEIP。
终端设备手动tracert到www.cloudflare.com，看是否路由跟踪到PPGW。
这一切都成立的话，可以打开浏览器访问`https://www.cloudflare.com/cdn-cgi/trace/`，或者使用curl命令（排除浏览器缓存等问题）`curl -v https://www.cloudflare.com/cdn-cgi/trace/`，同样可以观察走了节点IP，同时在虚拟机终端也会产生了连接日志。

**Reply by MyIsZero** [reply]
```
nslookup www.cloudflare.com
服务器:  PaoPaoDNS,blog.03k.org
Address:  192.168.0.250

名称:    www.cloudflare.com
Address:  7.0.0.8
```
```
tracert www.cloudflare.com

通过最多 30 个跃点跟踪
到 www.cloudflare.com [7.0.0.8] 的路由:

  1     2 ms     2 ms     2 ms  192.168.0.1
  2     3 ms     6 ms     2 ms  7.0.0.8

跟踪完成。
```
```
curl -v https://www.cloudflare.com/cdn-cgi/trace/
*   Trying 7.0.0.8:443...
* connect to 7.0.0.8 port 443 failed: Timed out
* Failed to connect to www.cloudflare.com port 443 after 21040 ms: Couldn't connect to server
* Closing connection 0
curl: (28) Failed to connect to www.cloudflare.com port 443 after 21040 ms: Couldn't connect to server
```
浏览器无法访问，PaoPaoGateWay里面也没任何反应，
```
https://www.cloudflare.com/cdn-cgi/trace/
```
socks5节点，我直接使用socks5不经过paopao是可以正常访问的，网页结果是ip=66.

**Reply by kkkgo** [reply]
你直接用socks5不是不经过dns/gw，是不经过路由器/静态路由。这说明你的静态路由/路由交换可能存在问题。
windows的话你可以直接使用管理员`route add 7.0.0.0 mask 255.0.0.0 192.168.0.251`手动添加静态路由来测试。

**Reply by kkkgo** [reply]
另外提一句，你用的是否是原版核心，你可能替换了存在问题的核心

**Reply by MyIsZero** [reply]
用的都是下载好的，没改什么核心

**Comment by MyIsZero** [comment]
这几天重新去买了个路由器，然后 论坛www.right.com.cn学刷机，把刚买的路由器刷了，好像可以用了，就是不知道PaoPaoGateWay怎么，有时候会刷屏
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/5971539d-0e67-41c9-816b-e9ddbe693c8f)
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/819c8a72-457d-4668-a306-53109dc062c6)

**Reply by MyIsZero** [reply]
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/cc02805e-28e9-468b-a3b9-cd0bafed1090)

**Reply by MyIsZero** [reply]
udp_enable的影响？

**Reply by kkkgo** [reply]
你这有国内流量跑过去的那就该考虑你的拓扑是不是搞错了。

**Comment by MyIsZero** [comment]
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/90bebc09-486d-4bea-b090-5f72346d2da1)
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/27af9ebd-2304-422b-b9b3-a18ed5258fc9)
使用socks匹配match()，MATCH 全匹配意思？
还会刷无法分配所请求的地址，确实这个报错会无法上google,但刷新网页又好了

**Reply by MyIsZero** [reply]
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/41cc4af2-f542-4a68-9a92-310da0dd5665)

**Reply by MyIsZero** [reply]
Telegram有时就会因为这报错卡住，重新获取下Telegram图片，又好了

**Comment by MyIsZero** [comment]
PaoPaoDNS（虚拟机docker）
```
docker run -d \
--name paopaodns \
--restart always \
-v /root/paopaodns:/data \
-e TZ=Asia/Shanghai \
-e DNS_SERVERNAME=PaoPaoDNS,blog.03k.org \
-e DNSPORT=53 \
-e CNAUTO=yes \
-e CNFALL=yes \
-e CN_TRACKER=yes \
-e IPV6=no \
-e SERVER_IP=192.168.0.250 \
-e CUSTOM_FORWARD=192.168.0.251:53 \
-e AUTO_FORWARD=no \
-e HTTP_FILE=yes \
-p 5333:53/tcp -p 5333:53/udp \
-p 5304:5304/tcp -p 5304:5304/udp \
-p 7889:7889/tcp \
sliamb/paopaodns
```
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/2d3df513-2654-4ba2-9a60-c533f5d9d2fc)
AdGuardHome去接127.0.0.1:5333

**Reply by MyIsZero** [reply]
```
#paopao-gateway

# mode=socks5|ovpn|yaml|suburl|free
# default: free
mode=socks5

# Set fakeip's CIDR here
# default: fake_cidr=7.0.0.0/8
fake_cidr=7.0.0.0/8

# Set your trusted DNS here
# default: dns_ip=1.0.0.1
dns_ip=192.168.0.250
# dns_ip=223.5.5.5
# default: dns_port=53
# If used with PaoPaoDNS, you can set the 5304 port
dns_port=5304
# dns_port=53

# Clash's web dashboard
clash_web_port="80"
clash_web_password="PaoPaoDNS0"

# default：openport=no
# socks+http mixed 1080
openport=no

# default: udp_enable=no
udp_enable=yes

# default:30
sleeptime=30

# socks5 mode settting
# default: socks5_ip=gatewayIP
socks5_ip="192.168.0.250"
# default: socks5_port="7890"
socks5_port="1080"

# ovpn mode settting
# The ovpn file in the same directory as the ppgw.ini.
# default: ovpnfile=custom.ovpn
ovpnfile="custom.ovpn"
ovpn_username=""
ovpn_password=""

# yaml mode settting
# The yaml file in the same directory as the ppgw.ini.
# default: yamlfile=custom.yaml
yamlfile="custom.yaml"

# suburl mode settting
suburl="https://..."
subtime=1d

# fast_node=check/yes/no
fast_node=no
test_node_url="https://www.youtube.com/generate_204"
# test_node_url="https://cp.cloudflare.com"
ext_node="Traffic|Expire| GB|Days|Date"
cpudelay="3000"

# dns burn setting
# depend on fast_node=yes & mode=suburl/yaml
dns_burn=no
# If used with PaoPaoDNS, you can set the PaoPaoDNS:53
ex_dns="192.168.0.250:5333"

# Network traffic records
net_rec=yes
max_rec=5000
```

**Comment by kkkgo** [comment]
1、建议检查“192.168.0.250”这个机器的DNS和网关的设置。
2、建议把出口节点直接放在ppgw里面避免不可预知的死循环。
3、Cannot assign requested address是连接数爆满的表现，说明有不该进入ppgw的流量进入了ppgw，或者拓扑方式不对，或者无法处理连接。

**Reply by MyIsZero** [reply]
PaoPaoDNS（192.168.0.250）使用的是dhcp，dns是192.168.0.250，虽然我想把节点配置放在ppgw.ini里面，但yaml就是用不了，放弃使用，在PaoPaoDNS（192.168.0.250）里面搞socks5，至于不该进入ppgw的流量，由于使用的是白名单域名，没理解有什么是不该的，域名都是我一个个手加的，拓扑方式嘛，也不是很理解哪不对，手机访问www.google.com，先和dns交互，然后定位跑到PaoPaoGateWay，PaoPaoGateWay再去连socks5

**Reply by kkkgo** [reply]
主要看是否有设备的DNS/网关被直接设置为了ppgw的IP，比如192.168.0.250的DNS/网关是192.168.0.251的话，这样就可能会引起大量连接数导致资源被耗尽。如果是虚拟机本身性能不足以处理，可以考虑增加内存。当然，启用UDP的处理也是比较耗费资源的。
yaml用不了要看你的协议是否被核心支持，若不支持可以通过docker定制更换clash核心。同时也可以下载clash的二进制命令行exe手动测试yaml配置是否可以用。
按你的描述把节点放在也DNS所在的机器上，那么需要注意的是节点域名本身是否被解析成fakeip，以及如果宿主机连接数不足可能会遇到问题。

**Reply by MyIsZero** [reply]
```
-e CUSTOM_FORWARD=192.168.0.251:53 \
-e AUTO_FORWARD=no \
```
节点应该是不会被解析成fakeip的，我在PaoPaoDNS里面设置了hosts，force_forward_list.txt里面我也没设节点的，观察了一整天的日志，感觉正常的日志是
```
[TCP] 192.168.0.112:39896 --> play-fe.googleapis.com:443 match Match() using socks
```
而不正常的刷屏日志是
```
[UDP] 127.0.0.1:49789 —>172.217.169.86:443 match Match() using socks
```
测试手机在打开手机play商店时，从127.0.0.1 using socks连接的日志就会刷屏，按照我观察的日志来理解，正常转到socks5的日志组成应该是访问来源ip--》www.google.com这种组成，但这种刷屏日志却从PaoPaoGateWay发出的，把socks5刷宕机了，没太理解打开手机play商店怎么访问来源是127.0.0.1，又使用手机浏览器测试
```
[TCP] 192.168.0.112:37070 --> www.google.com:443 match Match() using socks
```
正常的日志，没理解play商店是什么情况

**Reply by kkkgo** [reply]
感觉你去docker定制一个带嗅探sniff=yes的镜像可以改善下。

**Reply by MyIsZero** [reply]
```
docker pull sliamb/ppgwiso
docker run --name paopaoGateway --rm -e SNIFF=yes -v .:/data sliamb/ppgwiso
```
还是刷，电脑网页版的play商店不会，手机的play商店就会，手机打开play商店网页版不会刷屏
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/f69e5c15-2cc9-4505-a3bd-fba955378ec6)

**Reply by kkkgo** [reply]
开fast_node试试。

**Reply by MyIsZero** [reply]
fast_node=check，测试时是这样的设置的

**Reply by kkkgo** [reply]
开yes。

**Reply by MyIsZero** [reply]
fast_node=yes，用不了了。
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/e69a40a9-f92b-449a-9749-3343448275bf)
fast_node=no，fast_node=check，可以使用访问www.google.com是正常的
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/e71bb442-a359-401d-b481-00250de2c563)

**Reply by MyIsZero** [reply]
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/b6193d1a-4b78-4401-97c3-f0bd4212b7e2)
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/44c927e0-5ed0-47c5-9986-afbdc0ae9fc0)

**Reply by MyIsZero** [reply]
![图片](https://github.com/kkkgo/PaoPaoGateWay/assets/113719298/ca9e2a4f-ddd4-4ab1-b830-484d7c695fb5)

**Reply by MyIsZero** [reply]
好像是我socks5的配置问题，不过虽然不刷屏了，但fast_node=yes，还是用不了，继续fast_node=no

**Reply by kkkgo** [reply]
1、还是建议研究下yaml配置，把节点直接配置到ppgw里可以减少故障点和出现的问题，提高连接性能。
2、可以使用最新发布的镜像部署再观察日志。

**Comment by MyIsZero** [comment]
关于本篇中的路由跳转问题：
前提概要：
A路由器和B路由器是使用无线桥接，路由器上配置路由跳转后发现通讯有问题，B路由器下虚拟机的window配上路由就一切正常，但A下的设备还是通讯问题。
可参考：
https://www.right.com.cn/forum/thread-8295596-1-1.html
https://www.right.com.cn/forum/thread-206897-1-1.html
٩(╬ʘ益ʘ╬)۶
我要砸了这个路由器
